XamlControlsGallery/ControlPages/GridViewPage.xaml.cs

```diff
@@ -9,94 +9,504 @@
                         break;
                     case "Multiple":
-                        Control1.SelectionMode = ListViewSelectionMode.Multiple;
+                        ContentGridView.SelectionMode = ListViewSelectionMode.Multiple;
                         break;
                     case "Extended":
-                        Control1.SelectionMode = ListViewSelectionMode.Extended;
+                        ContentGridView.SelectionMode = ListViewSelectionMode.Extended;
                         break;
                 }
             }
         }
+
+        private void StyledGrid_IncreaseColSpace(object sender, RoutedEventArgs e)
+        {
+            ActualColSpace += 1;
+
+            // Update text box with newly increased value
+            ColSpace.Text = ActualColSpace.ToString();
+            ColSpace.PlaceholderText = ActualColSpace.ToString();
+
+            // Increase left/right margin on all GridViewItems
+            for (int i = 0; i < StyledGrid.Items.Count; i++)
+            {
+                GridViewItem item = StyledGrid.ContainerFromIndex(i) as GridViewItem;
+
+                // Create a Thickness property to bind to each item's Margin property.
+                // Use existing margin and only change relevant values (left/right)
+                Thickness NewMargin = item.Margin;
+                NewMargin.Left = item.Margin.Left + 1;
+                NewMargin.Right = item.Margin.Right + 1;
+
+                item.Margin = NewMargin;
+            }
+
+            NotifyPropertyChanged();
+        }
+
+        private void StyledGrid_DecreaseColSpace(object sender, RoutedEventArgs e)
+        {
+            // Make sure new value is positive
+            if (ActualColSpace - 1 >= 0)
+            {
+                ActualColSpace -= 1;
+
+                // Update text box with newly increased value
+                ColSpace.Text = ActualColSpace.ToString();
+                ColSpace.PlaceholderText = ActualColSpace.ToString();
+
+                // Increase left/right margin on all GridViewItems
+                for (int i = 0; i < StyledGrid.Items.Count; i++)
+                {
+                    GridViewItem item = StyledGrid.ContainerFromIndex(i) as GridViewItem;
+
+                    // Create a Thickness property to bind to each item's Margin property.
+                    // Use existing margin and only change relevant values (left/right)
+                    Thickness NewMargin = item.Margin;
+                    NewMargin.Left = item.Margin.Left - 1;
+                    NewMargin.Right = item.Margin.Right - 1;
+
+                    item.Margin = NewMargin;
+                }
+            }
+
+            // If new value is negative, reset text box to 0 and display an error message.
+            else
+            {
+                ColSpace.Text = "";
+                ColSpace.PlaceholderText = "0";
+                sendMessage();
```

Comment:
The MessageDialog will say "Value cannot be less than 1." which is not true for the column and row spacing. So instead, we should state that the value can not be negative (or prevent the scenario altogether).